### PR TITLE
Passing neon options to the console

### PR DESCRIPTION
**Pull request by prepor**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/proxy.rs [hunk 1]
```diff
@@ -945,3 +951,26 @@ impl<S: AsyncRead + AsyncWrite + Unpin> Client<'_, S> {
...
+        .intersperse(" ") // TODO: use impl from std once it's stabilized
+        .collect();
+
+    // Don't even bother with empty options.
+    if options.is_empty() {
+        return None;
+    }
+
+    Some(options)
+}
```
Comment (conradludgate) [inline proxy/src/proxy.rs:969]: dupe of other neon_options fn but unsorted? I would prefer a single function that always sorts - the cost of sorting is cheap for small lists (we should probably put a limit on large number of options)

**Comment by kelvich** [comment]
I'd suggest to do RFC on the options / branch-based routing / potential changes to branch naming and commit to some plane there. Here iterative approach could be exhausting since we will have to support any intermediate solutions potentially forever.

Also unit tests would be nice :)

**Comment by prepor** [comment]
> options / branch-based routing / potential changes to branch naming and commit to some plan there

Shouldn't be opaque for proxy? Options list, their names and meaning, hostname meaning and routing rules are part of `--auth-backend`

I of course for discussions of these things, but probably it's out of scope of proxy and this PR.

**Comment by kelvich** [comment]
Ah, fair. I though that this patch imposes syntax.

**Comment by prepor** [comment]
I moved this PR to https://github.com/neondatabase/neon/pull/5781, to not fork

**Pull request closed without merging**